src/passes/Metrics.cpp

```diff
@@ -62,16 +61,19 @@ struct Metrics
     counts["[tags]"] = imports.getNumDefinedTags();
     counts["[exports]"] = module->exports.size();
     counts["[tables]"] = imports.getNumDefinedTables();
-    // add memory and table
-    if (module->memory.exists) {
-      Index size = 0;
-      for (auto& segment : module->memory.segments) {
-        size += segment.data.size();
-      }
-      counts["[memory-data]"] = size;
-    }
 
+    // add memory
+    walkMemory(&module->memory);
+    for (auto& segment : module->dataSegments) {
+      walkDataSegment(segment.get());
+    }
     Index size = 0;
+    ModuleUtils::iterActiveDataSegments(
+      *module, [&](DataSegment* segment) { size += segment->data.size(); });
```

Comment:
I guess we already have this pattern established for element segments. It might be nice to do a drive-by cleanup here and use just a single loop each for element segments and data segments.